include/deal.II/lac/trilinos_sparse_matrix.h

```diff
@@ -2223,10 +2225,33 @@ namespace TrilinosWrappers
         * inverse operations.
         * The <tt>vmult</tt> and <tt>Tvmult</tt> operations retain the standard
         * definitions inherited from @p op.
+        *
+        * @note This function is enabled only if the solver and preconditioner
+        * derive from the respective TrilinosWrappers base classes.
         */
         template <typename Solver, typename Preconditioner>
-        TrilinosPayload inverse_payload (Solver &, const Preconditioner &) const;
+        typename std::enable_if<
+        std::is_base_of<TrilinosWrappers::SolverBase,Solver>::value and
```

Comment:
No, you're right as we do this in [linear_operator.h](https://github.com/dealii/dealii/blob/master/include/deal.II/lac/linear_operator.h#L240), for example. 

I happen to find the exact opposite - in the template craziness the text form is easier for me to see with syntax highlighting. But I'm very happy to stick with the standardised and widely adopted syntax. It didn't cross my mind to check this before opening a PR.